src/components/Pressable.tsx

```diff
@@ -0,0 +1,281 @@
+import React, { useRef } from 'react';
+import {
+  View,
+  AccessibilityProps,
+  ColorValue,
+  Insets,
+  NativeSyntheticEvent,
+  StyleProp,
+  TargetedEvent,
+  ViewProps,
+  ViewStyle,
+  StyleSheet,
+} from 'react-native';
+
+import {
+  Gesture,
+  GestureDetector,
+  GestureHandlerRootView,
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+  LongPressGestureHandlerEventPayload,
+  TouchData,
+} from '../.';
+import { HoverGestureHandlerEventPayload } from '../handlers/gestures/hoverGesture';
+
+const DEFAULT_LONG_PRESS_DURATION = 500;
+const DEFAULT_HOVER_DELAY = 0;
+
+export interface PressableStateCallbackType {
+  readonly pressed: boolean;
+}
+
+export interface PressableAndroidRippleConfig {
+  color?: null | ColorValue | undefined;
+  borderless?: null | boolean | undefined;
+  radius?: null | number | undefined;
+  foreground?: null | boolean | undefined;
+}
+
+export interface PressableProps
```

Comment:
I think it would make sense to split this component into multiple files, type definitions can be moved out